spark/src/test/scala/org/apache/spark/sql/delta/typewidening/TypeWideningInsertSchemaEvolutionSuite.scala

```diff
@@ -297,8 +297,7 @@ trait TypeWideningInsertSchemaEvolutionTests
           metadata = typeWideningMetadata(version = 1, from = ShortType, to = IntegerType)))))
   )
 
-  // The next two tests document inconsistencies when handling maps. Using SQL or INSERT by position
-  // doesn't allow type evolution but using dataframe INSERT by name does.
+  // maps now allow type evolution for INSERT by position and name in SQL and dataframe.
```

Comment:
It would be good to add more tests, for example in [DeltaInsertIntoSQLSuite](https://github.com/delta-io/delta/blob/3a98b8a33a48416649d0844355acffceca4ae21c/spark/src/test/scala/org/apache/spark/sql/delta/DeltaInsertIntoTableSuite.scala#L44):
- add a test covering the initial issue from https://github.com/delta-io/delta/issues/3227 - essentially a version of the example you provide, except that I don't think we run examples as part of PR tests (I might be wrong)
- add tests covering inserting data into a map type using key/values that have (1) a different type than in the table, (2) more column than in the table with schema evolution enabled/disabled
- if supporting maps/arrays nested inside maps: add a test covering 2 level deep nested maps